### PR TITLE
Firebase setup

### PR DESCRIPTION
**Pull request by niamh-power**
Setup firebase with anonymous authentication, without info plist :P

### CODE REVIEW

**Review by rock3r — CHANGES_REQUESTED** [review]
Nice one, I'd only take out all the various `//new` comments from the codelab, then good to go

File: lib/rootview.dart [hunk 1]
```diff
@@ -19,6 +22,9 @@ class Page<T> {
   }
 }
 
+final googleSignIn = new GoogleSignIn();
```
Comment (fourlastor) [inline lib/rootview.dart]: can these two be injected as dependencies?

File: lib/rootview.dart [hunk 2]
```diff
@@ -50,6 +56,8 @@ class RootView extends StatelessWidget {
         items: tabBarItems
     );
 
+    _ensureLoggedIn();
```
Comment (fourlastor) [inline lib/rootview.dart:68]: is this method blocking? I know it _feels_ blocking but I think it just returns a `Future`

File: lib/rootview.dart [hunk 3]
```diff
@@ -8,6 +8,9 @@ import 'package:flutter/material.dart';
 import 'package:SquanchyFlutter/favourites/favourites.dart';
 import 'package:SquanchyFlutter/schedule/schedule.dart';
 import 'package:SquanchyFlutter/venue/venuedetails.dart';
+import 'package:google_sign_in/google_sign_in.dart';                   // new
+import 'package:firebase_auth/firebase_auth.dart';                // new
```
Comment (fourlastor) [inline lib/rootview.dart]: I think the `// new` can go 👅

**Review by fourlastor — COMMENTED** [review]
Niiiiice! Only few comments 💃

File: lib/venue/venuedetails.dart [hunk 4]
```diff
@@ -9,6 +10,10 @@ class VenueDetailsView extends StatelessWidget {
   {
     final theme = Theme.of(context);
 
+    Firestore.instance.collection('conference_info').getDocuments().asStream().listen(
```
Comment (xrigau) [inline lib/venue/venuedetails.dart]: have you considered wrapping this with a service so that you can swap Firestore with whatever in the future?

File: lib/rootview.dart [hunk 5]
```diff
@@ -19,9 +21,22 @@ class Page<T> {
   }
 }
 
-class RootView extends StatelessWidget {
+final googleSignIn = new GoogleSignIn();
+final auth = FirebaseAuth.instance;
```
Comment (xrigau) [inline lib/rootview.dart:25]: are these being used?

File: ios/Runner.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -387,7 +387,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
```
Comment (rock3r) [inline ios/Runner.xcodeproj/project.pbxproj]: Are these changes intended?

File: ios/Podfile.lock [hunk 7]
```diff
@@ -1,16 +1,156 @@
...
+    :path: /Users/niamhpower/.pub-cache/git/flutter_plugins-9cabcc57e63fbc9be4fafee08e6435a1a8c71d61/packages/cloud_firestore/ios
+  firebase_analytics:
+    :path: /Users/niamhpower/.pub-cache/hosted/pub.dartlang.org/firebase_analytics-0.3.0/ios
+  firebase_auth:
+    :path: /Users/niamhpower/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.5.1/ios
   Flutter:
-    :path: /Users/takhion/flutter/bin/cache/artifacts/engine/ios
+    :path: /Users/niamhpower/flutter/bin/cache/artifacts/engine/ios
+  google_sign_in:
+    :path: /Users/niamhpower/.pub-cache/hosted/pub.dartlang.org/google_sign_in-3.0.0/ios
```
Comment (florianmski) [inline ios/Podfile.lock:126]: Those paths look sketchy 😄

Comment (rock3r) [reply ios/Podfile.lock:126]: #iostooling

**Review by rock3r — DISMISSED** [review]
Good for me now. @fourlastor ?

**Comment by rock3r** [comment]
Are we all ok with shipping this? @fourlastor @xrigau @florianmski — I think there's still a pending comment or two for @niamh-power

**Comment by fourlastor** [comment]
yeah I think it's good! just pending a clarification on the blocking (or not) blocking method

**Review by fourlastor — APPROVED** [review]

**Comment by niamh-power** [comment]
Yeah sorry I've been neglecting this, will have a look this evening

**Review by xrigau — APPROVED** [review]
my comments aren't blocking either 👍

File: ios/Podfile.lock (hunk 7 above)
Comment (niamh-power) [reply ios/Podfile.lock:126]: Yeah so the podfile.lock should be in version control, but something flutter does here appears to add these paths 🤷‍♀️ https://github.com/flutter/flutter/issues/9299

**Comment by niamh-power** [comment]
Not sure what's up with the CI, as it's building fine locally
